Zend/zend_compile.c

```diff
@@ -5005,18 +5005,18 @@ static zend_uchar determine_switch_jumptable_type(zend_ast_list *cases) {
 	return common_type;
 }
 
-static zend_bool should_use_jumptable(zend_ast_list *cases, zend_uchar jumptable_type) {
+static zend_bool should_use_jumptable(uint32_t num_children, zend_uchar jumptable_type) {
 	if (CG(compiler_options) & ZEND_COMPILE_NO_JUMPTABLES) {
 		return 0;
 	}
 
 	/* Thresholds are chosen based on when the average switch time for equidistributed
 	 * input becomes smaller when using the jumptable optimization. */
 	if (jumptable_type == IS_LONG) {
-		return cases->children >= 5;
+		return num_children >= 5;
 	} else {
 		ZEND_ASSERT(jumptable_type == IS_STRING);
-		return cases->children >= 2;
+		return num_children >= 2;
```

Comment:
Did you check whether this heuristic is reasonable for match? They probably have slightly different performance characteristics.

Reply:
I didn't, I assumed. I'll run a few benchmarks to make sure these values apply here.